buildAndReleaseTask/task.json

```diff
@@ -79,6 +79,14 @@
             "defaultValue": "Automated Release $(Release.ReleaseId)",
             "required": true,
             "helpMarkDown": "The message that is attached to the commit"
+        },
+        {
+            "name": "cleanRepository",
+            "type": "boolean",
+            "label": "Clean Repository?",
+            "defaultValue": "false",
```

Comment:
Based on schema, default Value handle boolean type.
https://github.com/Microsoft/azure-pipelines-task-lib/blob/master/tasks.schema.json#L171&L175

Reply:
Changed value to boolean instead of string